src/runtime_src/tools/scripts/peta_build.sh

```diff
@@ -0,0 +1,164 @@
+
+echo " * Configuring PetaLinux Project"
+# Allow users to access shell without login
+#echo "******************* configure petalinux project hook ********************"
+echo "CONFIG_YOCTO_ENABLE_DEBUG_TWEAKS=y" >> ${PETALINUX_NAME}/project-spec/configs/config
+echo "petalinux-config -p $PETALINUX_NAME --get-hw-description=${DSA_DIR} --oldconfig"
+petalinux-config -p $PETALINUX_NAME --get-hw-description=${DSA_DIR} --oldconfig
+
+echo " * Change to meta directory: ${PETALINUX_NAME}/project-spec/meta-user/"
+cd ${PETALINUX_NAME}/project-spec/meta-user/
+
+echo " * Copying ${RECIPE_DIR} to `readlink -f .`"
+cp -r ${RECIPE_DIR} .
+
+# If you are using PetaLinux 2018.3 or earlier version, do below step
+if [[ $PETALINUX_VER == "2018"* ]]; then
+  echo " * A 2018.X PetaLinux release (${PETALINUX_VER}) was detected, copying opencl-headers recipe:"
+  mkdir -p recipes-xrt/opencl-headers
+  wget -O recipes-xrt/opencl-headers/opencl-headers_git.bb http://cgit.openembedded.org/meta-openembedded/plain/meta-oe/recipes-core/opencl-headers/opencl-headers_git.bb
+fi
+
+#echo "******************* application hook ********************"
+# mnt-sd will run at the Linux boot time, it do below things
+#  1. mount SD cart to /mnt
+#  2. run init.sh in /mnt if it exist
+cp -r ${RECIPE_DIR}/../mnt-sd recipes-apps/
+
+echo " * Adding XRT, HAL, Driver recipes"
+
+# In 2018.3 Petalinux the name of this file changed..
+if [ -f recipes-core/images/petalinux-image.bbappend ]; then
+	PETALINUX_IMAGE_BBAPPEND=recipes-core/images/petalinux-image.bbappend
+elif [ -f recipes-core/images/petalinux-image-full.bbappend ]; then
+	PETALINUX_IMAGE_BBAPPEND=recipes-core/images/petalinux-image-full.bbappend
+else
+	echo "Not petalinux image .bbappend file in project-spec/meta-user/recipes-core/images/"
+	exit 1;
+fi
+
+#echo "******************* petalinux-image hook ********************"
+echo 'IMAGE_INSTALL_append = " xrt-dev"'            >> $PETALINUX_IMAGE_BBAPPEND
+echo 'IMAGE_INSTALL_append = " mnt-sd"'             >> $PETALINUX_IMAGE_BBAPPEND
+echo 'IMAGE_INSTALL_append = " xrt"'                >> $PETALINUX_IMAGE_BBAPPEND
+echo 'IMAGE_INSTALL_append = " zocl"'               >> $PETALINUX_IMAGE_BBAPPEND
+echo 'IMAGE_INSTALL_append = " opencl-headers-dev"' >> $PETALINUX_IMAGE_BBAPPEND
+echo 'IMAGE_INSTALL_append = " opencl-clhpp-dev"'   >> $PETALINUX_IMAGE_BBAPPEND
+
+echo " * Adding XRT Kernel Node to Device Tree"
+#echo "******************* Device Tree hook ********************"
+echo "cat ${RECIPE_DIR}/../../runtime_src/core/edge/fragments/xlnk_dts_fragment_mpsoc.dts >> recipes-bsp/device-tree/files/system-user.dtsi"
+cat ${RECIPE_DIR}/../../runtime_src/core/edge/fragments/xlnk_dts_fragment_mpsoc.dts >> recipes-bsp/device-tree/files/system-user.dtsi
+echo "cat ${ORIGINAL_DIR}/dsa_build/zcu102_fragment.dts >> recipes-bsp/device-tree/files/system-user.dtsi"
+cat ${ORIGINAL_DIR}/dsa_build/zcu102_fragment.dts >> recipes-bsp/device-tree/files/system-user.dtsi
```

Comment:
Hi Min,

peta_build.sh contains following lines which are specific to zcu102.dts addtion. We should give some extra option to specify the file which contains all the dts paths.

example: --dts-file <file which contains all extra dts files>

Reply:
Good eye! I will fix this in next commit.